sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java

```diff
@@ -66,7 +65,7 @@
      * BOM header from some response bodies. To be removed in deserialization.
      */
     private static final String BOM = "\uFEFF";
-    private static final String BOM_STRING = new String(BOM.getBytes(StandardCharsets.UTF_8), Charset.defaultCharset());
+    private static final String BOM_STRING = new String(BOM.getBytes(StandardCharsets.UTF_8), StandardCharsets.UTF_8);
```

Comment:
Do we need this BOM stripping on already stripped-and-decoded string anymore?

Reply:
Shouldn't need it anymore, will add tests into the share `RestProxyTests` that will validate encoding handling for all clients.

Reply:
Do we need to apply a similar BOM lookup here as well?

https://github.com/Azure/azure-sdk-for-java/blob/88adf363b4a073b9823bfa7fb939c18fffb68278/sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/BufferedHttpResponse.java#L63

Reply:
If so, wondering whether it makes sense to move this to FluxUtils?  something like: 

```java
Mono<String> FluxUtils.toBomAwarePublisher(
    Flux<ByteBuffer> stream,  
    String contentEncoding)
```

where `contentEncoding` is the value of Content-Encoding header, if no BOM prefix in-stream and if `contentEncoding` is null then method uses utf8 as default.

Reply:
Added helper into `CoreUtils`